calcom/cal.com
# fix/slots-calculate-hours

## What does this PR do?
Redo of how we calculate slots available per day, instead of iterating by each frequency available till the end of the day, we move to split working hours in chunks by X frequency and only iterate on those.

Fixes #1959 

## Type of change




## How should this be tested?

- Create an 60 min event, that runs on any day but one of its availability starts at 12:30 and ends at 13:30 and it should display correctly along with others availabilities that start on eg. 2:00-3:00, 15:00-16:00.

## Checklist
